girder/web_client/src/views/widgets/BrowserWidget.js

```diff
@@ -75,17 +82,24 @@ var BrowserWidget = View.extend({
             this.root = evt.root;
             this._renderHierarchyView();
         });
+        if (this.defaultSelectedResource && this.defaultSelectedResource.parent) {
+            this.root = this.defaultSelectedResource.parent;
```

Comment:
Just some planning for going about doing this:

So the original default was to have no root set and by default it would just present something that is not selected.  I think it is best to calculate the parent if we can to reduce friction from using the 'selectedDefaultResource' parameter with the browser widget.

Some Background Information for myself:
Item - Can get a `folderId` and fetch to get the parent model
Folder - parent is either a folder/collection/user: `parentCollection` ('folder'|'collection'|'user') and `parentId`, meaning I have to conditionally fetch each different type of model and set the root to that instead.  
**Question**: Looking through hHerarchyWidget it references the root (parentModel) 92 times, so I'm assuming through all the functions it needs an explicit version of the specific model type (Item,Collection,User,Folder) and not some generic parent?

My proposed process for this:
1. Check to see if a `root` is already assigned, if so default to that.
2. If a `defaultSelectedResource` is assigned and already has an existing `parent`, use that as the new root
3. If no `parent` exists determine if it has a `folderId` associated with the defaultSelectedResource, followed by creating a FolderModel and fetching the data, on fetch set the root to the FolderModel
4. If no folderId exists check for a `parentCollection` and `parentId`, depending on those values either create and fetch a `UserModel`, `CollectionModel`, or `FolderModel` then assign root to that
5. finally if none of the above exists we default to the undefined root

Additional Self Notes:
Now there are ways I should be able to generalize the Model retrieval and collection, they should all have `.on('g:fetched')` and on.('g:error').  Possibly an object with the `parentCollection` types as a key to the different Model constructors.
Also just default to using the collectionParent type folder, for if the folderId exists instead of duplicating code for retrieval of the model.

Reply:
folders all have baseParentId and baseParentType that will be a collection or user, so the `rootSelectorSettings` can be `{selectbyResource: <folder or item's folder>.get('baseParentId')}`.

Reply:
Because of the virtual folders plugin, it is useful to be able to provide the intended parent.  Having it default to the item's `folderId` would be handy use cases which aren't virtual-folder aware, though.  Since `item.parent` is used for access control, I'm not sure of the repercussions of overloading it to do this.  It might be better to have another option for this.

Reply:
> folders all have baseParentId and baseParentType that will be a collection or user, so the `rootSelectorSettings` can be `{selectbyResource: <folder or item's folder>.get('baseParentId')}`.

Yes for the rootSelectorWidget I could just use that, but I do need the parent of the current model for the HierarchyWidget. In the master branch the hierarachyWidget within browserWidget uses the root as the parentModel, for displaying the folder and the breadcrumb trail of the current folder.
![image](https://user-images.githubusercontent.com/61746913/77076793-b3dae980-69ca-11ea-8a89-173911428feb.png)

Before I don't think this was an issue because the root was a collection/user.  Now it needs to be a collection or folder to open to the proper location initially.  I can change this in the browserWidget to use my own value instead of `this.root` when a selectedDefaultResource Model is available.

Reply:
Agreed.  The `baseParentId` only helps the `rootSelectorSettings`, not the HierarchyWidget's parentModel.